### PR TITLE
Implement new builder and evaluation pipelines

### PR DESCRIPTION
**Pull request by remg1997**
### Description:
- Implements new builder and evaluation pipelines. 

### Key features:
1. Builder now deploys the models using ECS Fargate for heavy tasks (inference on whole datasets for benchmarks)
2. Builder is compatible with Dynalab 2.0
3. New ORM with updated repositories
4. Compatible evaluation pipeline for the new builder

### Pending features:
1. Test suite for builder and evaluation
2. Decentralized solution for communication with the database
3. Batch inference adapter for evaluation pipeline (requires batch inference implementation on Dynalab 2.0) 
4. Compute memory usage and throughput metrics
5. API Gateway integration with AWS Lambda for light tasks (inference on adversarial samples)

### CODE REVIEW

**Comment by douwekiela** [comment]
Adding fastapi to the pyproject config should fix the linter I think

File: new_builder_evaluation/app/domain/evaluation_decentralized.py [hunk 1]
```diff
@@ -0,0 +1,323 @@
...
+import boto3
+import jsonlines
+import requests
+
+from app.domain.builder import Builder
+from app.domain.eval_utils.evaluator import Evaluator
+from app.domain.eval_utils.input_formatter import InputFormatter
+
+
+class Evaluation:
```
[the remark this answers is not in the document]
Comment (Ciroye) [reply new_builder_evaluation/app/domain/evaluation_decentralized.py:22]: The idea is that we are going to create a PR focused 100 percent on decentralized so we are going to address all these issues in that PR

File: new_builder_evaluation/app/api/endpoints/evaluation.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+
+@router.get("/get_task_configuration")
+def get_task_configuration(id: int) -> dict:
+    evaluation = Evaluation()
+    return evaluation.get_task_configuration(id)
+
+
+@router.get("/get_model_id_and_task_code")
+def get_model_id_and_task_code(task_id: int):
```
[the remark this answers is not in the document]
Comment (Ciroye) [reply new_builder_evaluation/app/api/endpoints/evaluation.py:36]: The idea is that we are going to create a PR focused 100 percent on decentralized so we are going to address all these issues in that PR

File: new_builder_evaluation/app/domain/evaluation_decentralized.py [hunk 3]
```diff
@@ -0,0 +1,323 @@
...
+            aws_secret_access_key=os.getenv("AWS_SECRET_ACCESS_KEY"),
+            region_name=os.getenv("AWS_REGION"),
+        )
+        self.s3 = self.session.client("s3")
+        self.sqs = self.session.client("sqs")
+        self.s3_bucket = os.getenv("AWS_S3_BUCKET")
+        self.builder = Builder()
+
+    def require_fields_task(self):
+        spec = importlib.util.spec_from_file_location("ModelSingleInput", "./test.py")
```
[the remark this answers is not in the document]
Comment (Ciroye) [reply new_builder_evaluation/app/domain/evaluation_decentralized.py:35]: We don't know why, but not all tasks specify all the inputs, so we decided to use Dynalab 2.0 to verify

File: new_builder_evaluation/app/domain/evaluation.py [hunk 4]
```diff
@@ -0,0 +1,309 @@
...
+    def validate_input_schema(self, schema: list, dataset: list):
+        for param in schema:
+            if not (all([param in d for d in dataset])):
+                raise Exception(f"Missing param: {param}")
+
+    def build_single_request(self, schema: list, sample_dataset: dict):
+        return {param: sample_dataset.get(param) for param in schema}
+
+    def heavy_prediction(self, datasets: list, task_code: str, model: str):
+        folder_name = model.split("/")[-1].split(".")[0]
```
[the remark this answers is not in the document]
Comment (Ciroye) [reply new_builder_evaluation/app/domain/evaluation.py]: we are using a pattern which is the user id plus the model name. I think for the next version it is better to keep only the model name.

**Review by Ciroye — APPROVED** [review]

**Pull request merged**